django/db/models/functions/datetime.py

```diff
@@ -12,31 +12,60 @@
 from django.utils.functional import cached_property
 
 
-class Extract(Transform):
+class TimezoneMixin(object):
+    tzinfo = None
+
+    def get_tzname(self):
+        # Timezone conversions must happen to the input datetime *before*
+        # applying a function. 2015-12-31 23:00:00 -02:00 is stored in the
+        # database as 2016-01-01 01:00:00 +00:00. Any results should be
+        # based on the input datetime not the stored datetime.
+        tzname = None
+        if settings.USE_TZ:
+            if self.tzinfo is None:
+                tzname = timezone.get_current_timezone_name()
+            else:
+                tzname = timezone._get_timezone_name(self.tzinfo)
+        return tzname
+
+
+class Extract(Transform, TimezoneMixin):
```

Comment:
Unless there's a style reason there's no need to put it to the left. The mixin isn't overriding anything so the MRO isn't an issue. I find the main class (Transform in this case) being on the left more readable.